### PR TITLE
Wallbox set maximum charging current as number

### PR DESCRIPTION
**Pull request by bullitt186**
Based on the discussion in issue https://github.com/torbennehmer/hacs-e3dc/issues/47 and requested by @Wolfgang-03,
This PR implements the functionality to set the maximum charging current for a wallbox via a number Entity (Which allows sliders without helpers).

Under the hood, this PR brings the following features:
- Introduces E3DCNumber which allows to call methods on-change. (Will be handy for setting other E3DC values in the future)
- Reads & respects the hard upper and lower current limits as stored in the wallbox. These limits can only be set by the E3DC electrician during installation, so this is now more fail-safe compared to the previously hardcoded limits. 

For reference, see the two videos:
First one shows how by changing the number, the sensor which reads back from the E3DC the set current updates.
https://github.com/user-attachments/assets/595fbdda-e6d0-4a9e-b422-128d929c4a8b

2nd one shows how Number as well as Sensor update, when changing the max charging current from somewhere else
(In this case i had a 2nd HA instance running on another machine and changed from there max charging current. 
https://github.com/user-attachments/assets/919f8ad6-c7d0-4afc-b6b9-88569befd37b

### CODE REVIEW

**Comment by Wolfgang-03** [comment]
Looks great. I don't know if I'll be able to test it today and tomorrow morning I'm going away for the weekend. But next week I will definitely test it extensively.

File: custom_components/e3dc_rscp/coordinator.py [hunk 1]
```diff
@@ -136,6 +138,29 @@ def wallboxes(self, value: list[dict[str, str | int]]) -> None:
         """Set the list of wallboxes."""
         self._wallboxes = value
 
+    # Setter for individual wallbox values
+    def setWallboxValue(self, index: int, key: str, value: Any) -> None:
+        """Set the value for a specific key in a wallbox identified by its index."""
+        for wallbox in self._wallboxes:
+            if wallbox['index'] == index:
+                wallbox[key] = value
```
Comment (torbennehmer) [inline custom_components/e3dc_rscp/coordinator.py:146]: @bullitt186 do we want this as generic function like this? I normally prefer typed specific accessors, they make for cleaner code ad catch errors early.

File: custom_components/e3dc_rscp/number.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    def _handle_coordinator_update(self) -> None:
+        """Process coordinator updates."""
+        self._attr_value = self.coordinator.data.get(self.entity_description.key)
+        self.async_write_ha_state()
+
+    async def async_set_native_value(self, value: float) -> None:
+        """Set the number value asynchronously."""
+        if self.entity_description.async_set_native_value_action is not None:
+            self._attr_value = value
+            self.async_write_ha_state()
```
Comment (torbennehmer) [inline custom_components/e3dc_rscp/number.py:104]: can you await this? If so, please do so. Same above in handle_coordinator_update. HA uses a lot of async/await, up to throwing errors/warning if this gets mixed up

**Review by torbennehmer — CHANGES_REQUESTED** [review]

**Comment by torbennehmer** [comment]
@bullitt186 you don't have to add the changes of the other commit into this branch, they're already in main, git will take care of this. I suspect, in this situation we might actually get a conflict when merging  (haven't checked yet though). You can either leave it out or rebase to latest main, both will work from a Git's perspective. As long as this PR doesn't touch the device detection, leave it out, that's easiest (i.e. hard reset to one before HEAD, force push).

File: custom_components/e3dc_rscp/number.py (hunk 2 above)
Comment (bullitt186) [reply custom_components/e3dc_rscp/number.py:104]: This unfortunately results in this error:
"TypeError: object NoneType can't be used in 'await' expression"

I googled this a little bit and it seems to be the common way to not await async_write_ha_state()

File: custom_components/e3dc_rscp/coordinator.py (hunk 1 above)
Comment (bullitt186) [reply custom_components/e3dc_rscp/coordinator.py:146]: Good point. And it is not being used in any situation at all (and probably never will be, as wallboxes are only added by the coordinator itself.)
I remove it.

**Comment by bullitt186** [comment]
I included your feedback and rebased this one to main, IMHO it should merge just fine.

**Review by torbennehmer — APPROVED** [review]

**Pull request merged**